### PR TITLE
Enabled TinyTDS driver for Slacker

### PR DESCRIPTION
**Pull request by erickangMSFT**
This PR is to enable tiny_tds driver support for slacker to support macOS, Linux and Windows developers. 

Changelog

1. Added require'tiny_tds'  s.add_dependency 'tiny_tds', '~>2.0'
2. updated   s.add_dependency 'ruby-odbc', '= 0.99998' to prevent gem install slacker failure in macOS
3. added BigDecimal and Time match_values in query_result_matcher to support datetime2 format and bigdecimal comparison in multiple format including scientific notation
4. added BigDecimal and Time match unittest in spec/query_matcher_spec.rb
5. added two properties in database.yaml. port to support different port than 1433. driver to switch between 'odbc' and 'tiny_tds'
6. selecting 'odbc' driver type should make slacker compatible with existing project. 
7. tested on macOS and Linux ubuntu 16.04, ruby 2.4.1
8. updated the version number to 1.0.16 in slacker.gemspec

### CODE REVIEW

**Comment by jdshkolnik** [comment]
FYI, @vassilvk is on vacation until 8/23.

File: lib/slacker/application.rb [hunk 1]
```diff
@@ -25,12 +26,18 @@ class Application
 set concat_null_yields_null on;
 EOF
 
+    ODBC_Driver = 'odbc'
```
Comment (vassilvk) [inline lib/slacker/application.rb]: Please use Ruby's naming convention for constants (all-caps separated with underscores).
In this case the constant name should be ODBC_DRIVER.

File: lib/slacker/application.rb [hunk 2]
```diff
@@ -140,8 +157,27 @@ def configure_db
       end
     end
 
+    def configure_db_tiny_tds
+      begin
+        @database = TinyTds::Client.new :username => @configuration.db_user, :password => @configuration.db_password, 
+                    :host => @configuration.db_server, :database => @configuration.db_name, :port => 1433
```
Comment (vassilvk) [inline lib/slacker/application.rb]: I believe here you wanted to set the port to @configuration.db_port as opposed to hard-coding to 1433.

File: lib/slacker/application.rb (hunk 1 above)
Comment (erickangMSFT) [inline lib/slacker/application.rb]: Changed ODBC_Driver to ODBC_DRIVER to meet the naming convention.

File: lib/slacker/application.rb (hunk 2 above)
Comment (erickangMSFT) [inline lib/slacker/application.rb]: fixed the bug with :port => @configuration.db_port

File: lib/slacker/application.rb [hunk 3]
```diff
@@ -191,6 +191,7 @@ def query_script_odbc(sql)
         ensure
           st.drop unless st.nil?
       end
+      results.count > 1 ? results : results.first
```
Comment (erickangMSFT) [inline lib/slacker/application.rb]: In my changed, I introduced a bug not correctly returning the results. Fixed it and tested on Windows 10 with odbc driver.

File: spec/query_result_matcher_spec.rb [hunk 4]
```diff
@@ -256,7 +256,7 @@ def deep_copy(obj)
   describe 'DateTime2-based golden master' do
     before(:each) do
       @subject = [{'Field 1' => Time.parse('2017-01-01.000000'), 'Field_2' => 12}]
-      @correct_golden_master = Time.parse('2017-01-01 00:00:00 -0800')
+      @correct_golden_master = Time.parse('2017-01-01 00:00:00')
```
Comment (erickangMSFT) [inline spec/query_result_matcher_spec.rb]: fixed the test case failure on Windows.

**Pull request merged**